pkg/controller/controller.go

```diff
@@ -19,56 +33,595 @@ type Controller struct {
+	container := api.Container{
+		Name:            ContainerName,
+		Image:           containerImage,
+		ImagePullPolicy: api.PullAlways,
+	}
+	env := []api.EnvVar{
+		{
+			Name:  Namespace,
+			Value: b.Namespace,
+		},
+		{
+			Name:  TPR,
+			Value: b.Name,
+		},
+	}
+	for _, e := range env {
+		container.Env = append(container.Env, e)
+	}
+	container.Args = append(container.Args, "watch")
+	container.Args = append(container.Args, "--v=10")
+	backupVolumeMount := api.VolumeMount{
+		Name:      b.Spec.Destination.Volume.Name,
+		MountPath: b.Spec.Destination.Path,
+	}
+	sourceVolumeMount := api.VolumeMount{
+		Name:      b.Spec.Source.VolumeName,
+		MountPath: b.Spec.Source.Path,
+	}
+	container.VolumeMounts = append(container.VolumeMounts, backupVolumeMount)
+	container.VolumeMounts = append(container.VolumeMounts, sourceVolumeMount)
+	return container
+}
+
+func restartPods(kubeClient *internalclientset.Clientset, namespace string, opts api.ListOptions) error {
+	pods, err := kubeClient.Core().Pods(namespace).List(opts)
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		deleteOpts := &api.DeleteOptions{}
+		err = kubeClient.Core().Pods(namespace).Delete(pod.Name, deleteOpts)
+		if err != nil {
+			errMessage := fmt.Sprint("Failed to restart pod %s cause %v", pod.Name, err)
+			log.Println(errMessage)
+		}
+	}
+	return nil
+}
+
+func getKubeObject(kubeClient *internalclientset.Clientset, destination api.Volume, namespace string, ls labels.Selector) map[string]interface{} {
+	uslist := &runtime.UnstructuredList{}
+	err := kubeClient.Core().RESTClient().Get().Resource("replicationcontrollers").Namespace(namespace).LabelsSelectorParam(ls).Do().Into(uslist)
+	if err == nil && len(uslist.Items) != 0 {
+		return uslist.Items[0].Object
+	}
+
+	err = kubeClient.Extensions().RESTClient().Get().Resource("replicasets").Namespace(namespace).LabelsSelectorParam(ls).Do().Into(uslist)
+	if err == nil && len(uslist.Items) != 0 {
+		return uslist.Items[0].Object
+	}
 
+	err = kubeClient.Extensions().RESTClient().Get().Resource("deployments").Namespace(namespace).LabelsSelectorParam(ls).Do().Into(uslist)
+	if err == nil && len(uslist.Items) != 0 {
+		return uslist.Items[0].Object
+	}
+
+	err = kubeClient.Extensions().RESTClient().Get().Resource("daemonsets").Namespace(namespace).LabelsSelectorParam(ls).Do().Into(uslist)
+	if err == nil && len(uslist.Items) != 0 {
+		return uslist.Items[0].Object
+	}
+
+	err = kubeClient.Apps().RESTClient().Get().Resource("statefulsets").Namespace(namespace).LabelsSelectorParam(ls).Do().Into(uslist)
+	if err == nil && len(uslist.Items) != 0 {
+		return uslist.Items[0].Object
+	}
+	return nil
+}
+
+func findSelectors(lb map[string]string) labels.Selector {
+	set := labels.Set(lb)
+	selectores := labels.SelectorFromSet(set)
+	return selectores
+}
+
+func (pl *Controller) updateObjectAndStartBackup(b *rapi.Backup) error {
+	factory := cmdutil.NewFactory(nil)
+	kubeClient, err := factory.ClientSet()
```

Comment:
Why Not kubeclient stays inside the *Controller? and just use that instead of crating it everytime?